py/nlr.h

```diff
@@ -89,8 +89,12 @@
     #define MICROPY_NLR_MIPS (1)
     #define MICROPY_NLR_NUM_REGS (MICROPY_NLR_NUM_REGS_MIPS)
 #else
+    #ifndef MICROPY_NLR_SETJMP
```

Comment:
The idea here is to be able to say "I don't have support for setjmp" (by setting MICROPY_NLR_SETJMP to zero), in which case, if the arch-specific configuration cannot be done (because it falls through the if/elif), then don't try and enable setjmp.

So it works because like you say, the top-level if evaluates to true, but we need the final fall through to just set MICROPY_NLR_NUM_REGS to zero to disable NLR entirely.

I guess what we really need is a way for the port to convey that NLR is best-effort (i.e. if the arch-specific implementation is unavailable, it doesn't matter, but don't use setjmp). But really the better option would be to have a way to disable NLR. So I've done that instead.